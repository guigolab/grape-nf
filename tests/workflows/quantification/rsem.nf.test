nextflow_workflow {

    name "Test Workflow quantification"
    script "workflows/quantification/rsem.nf"
    workflow "quantification"

    tag "workflow"

    test("Should run quantification with RSEM") {

        when {
            params {
                stepList = [ "quantification" ]
            }
            workflow {
                """
                input[0] = file("${baseDir}/data/genome.fa")
                input[1] = file("${baseDir}/data/annotation.gtf")
                input[2] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE" ]
                ])
                input[3] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE" ]
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    // >>> TEST COMMENTED BECAUSE RSEM 1.3.3 FAILS WITH --calc-ci
    // >>> see https://github.com/deweylab/RSEM/issues/134
    // >>> DO NOT CLEAN SNAPSHOT!
    // test("Should run quantification with RSEM (with credibility intervals)") {

    //     when {
    //         params {
    //             stepList = [ "quantification" ]
    //             rsemCalcCI = true
    //         }
    //         workflow {
    //             """
    //             input[0] = file("${baseDir}/data/genome.fa")
    //             input[1] = file("${baseDir}/data/annotation.gtf")
    //             input[2] = Channel.from([
    //                 [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE" ]
    //             ])
    //             input[3] = Channel.from([
    //                 [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE" ]
    //             ])
    //             """
    //         }
    //     }

    //     then {
    //         assert workflow.success
    //         assert snapshot(workflow.out).match()
    //     }

    // }

}
