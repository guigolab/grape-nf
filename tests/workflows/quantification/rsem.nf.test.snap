{
    "Should run quantification with RSEM (with credibility intervals)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,568855cd154369988f0e9120660dcea6",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,a2ddfd7d7919830d69f941176ddff86f",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genes": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,568855cd154369988f0e9120660dcea6",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "isoforms": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,a2ddfd7d7919830d69f941176ddff86f",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T17:55:23.922489"
    },
    "Should run quantification with RSEM": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,d76a7103c9d815fa46dc7b3ab38ae4b1",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,2f825c2b612114534df653ec4c172c18",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genes": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,d76a7103c9d815fa46dc7b3ab38ae4b1",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "isoforms": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,2f825c2b612114534df653ec4c172c18",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T17:01:27.333102"
    }
}